doc/rm/asm_coding_style.md

```diff
@@ -176,12 +183,49 @@ _trap_start:
 
 ***Comments must use either the `//` or `/* */` syntaxes.***
 
-Every function-like label which is meant to be called like a function (*especially* `.globl`s) should be given a Doxygen-style comment.
+Every function-like label which is meant to be called like a function (*especially* `.global`s) should be given a Doxygen-style comment.
 While Doxygen is not suited for assembly, that style should be used for consistency.
 See the [C/C++ style guide]({{< relref "c_cpp_coding_style" >}}) for more information.
 
+Comments should be indented to match the line immediately after. For example:
+```S
+  // This comment is correctly indented.
+  call foo
+
+// This one is not.
+  call foo
+```
+
 All other advice for writing comments, as in the C/C++ style guide, also applies.
 
+
+### Declaring a Symbol
+
+***All "top-level" symbols must have the correct preamble and footer of directives.***
+
+To aid the disassembler, every function must follow the following template:
+
+```S
+  /**
+   * Comment describing what my function does
+   */
+  .section .some_section  // Optional if the previous symbol is in this setion.
+  .balign 4
+  .global my_function  // Only for exported symbols.
+  .type my_function, @function
+my_function:
+  // Instructions and stuff.
+  .size my_function, .-my_function
+```
+
+Note that `.global` is not spelled with the legacy `.globl` spelling. If the symbol
```

Comment:
Markdown style guide specifies one sentence per line for rendered files (for clearer diffs). I think we should stick to that here.

Reply:
I'll fix it in this PR, but we should revisit this rule... no one follows it in any other markdown in the tree, and rules people don't follow are not worth having.